src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.jenkins.support.filter;
+
+import com.cloudbees.jenkins.support.util.Persistence;
+import hudson.Extension;
+import hudson.model.AbstractItem;
+import hudson.model.ManagementLink;
+import hudson.model.Saveable;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.stream.Collectors.toConcurrentMap;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Holds all anonymized content mappings and provides a management view to see those mappings.
+ *
+ * @since TODO
+ */
+@Restricted(NoExternalUse.class)
+public class ContentMappings extends ManagementLink implements Saveable, Iterable<ContentMapping> {
+
+    /**
+     * @return the singleton instance
+     */
+    public static ContentMappings get() {
+        return all().get(ContentMappings.class);
+    }
+
+    /**
+     * Constructs a new ContentMappings using an existing config file or default settings if not found.
+     */
+    public static @Extension ContentMappings newInstance() throws IOException {
+        ContentMappings mappings = Persistence.load(ContentMappings.class);
+        if (mappings == null) {
+            mappings = (ContentMappings) new XmlProxy().readResolve();
+        }
+        return mappings;
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(ContentMappings.class.getName());
+
+    private final Set<String> stopWords;
+    private final Map<String, ContentMapping> mappings;
+
+    private ContentMappings(@Nonnull XmlProxy proxy) {
+        stopWords = proxy.stopWords == null ? getDefaultStopWords() : proxy.stopWords;
+        mappings = proxy.stopWords == null
+                ? new ConcurrentHashMap<>()
+                : proxy.mappings.stream().collect(toConcurrentMap(ContentMapping::getOriginal, Function.identity(), (a, b) -> {throw new IllegalArgumentException();}, ConcurrentSkipListMap::new));
```

Comment:
The mappings don't use the sort order defined in ContentMapping::COMPARATOR, they sort based on the order of the keys, so I am still seeing the problem where if I have a view named `Test` mapped to `Foo` and a view named `Prefix_Test` mapped to `Bar`, I see things like `Prefix_view_Foo` in the output. I was able to fix it with the following patch (probably worth adding a test though and maybe removing the comparable implementation from ContentMapping if it is unused):

    diff --git a/src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java b/src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java
    index 22c0684..ae9c488 100644
    --- a/src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java
    +++ b/src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java
    @@ -37,13 +37,13 @@ import javax.annotation.Nonnull;
     import java.io.IOException;
     import java.util.Arrays;
     import java.util.Collections;
    +import java.util.Comparator;
     import java.util.HashSet;
     import java.util.Iterator;
     import java.util.Locale;
     import java.util.Map;
     import java.util.Set;
     import java.util.Spliterator;
    -import java.util.concurrent.ConcurrentHashMap;
     import java.util.concurrent.ConcurrentSkipListMap;
     import java.util.function.Consumer;
     import java.util.function.Function;
    @@ -61,6 +61,10 @@ import static java.util.stream.Collectors.toMap;
     @Restricted(NoExternalUse.class)
     public class ContentMappings extends ManagementLink implements Saveable, Iterable<ContentMapping> {
     
    +    private static final Comparator<String> BY_LENGTH = Comparator.comparingLong(String::length);
    +    private static final Comparator<String> BY_NAME = Comparator.comparing(Function.identity());
    +    private static final Comparator<String> COMPARATOR = BY_LENGTH.reversed().thenComparing(BY_NAME);
    +
         /**
          * @return the singleton instance
          */
    @@ -87,8 +91,8 @@ public class ContentMappings extends ManagementLink implements Saveable, Iterabl
         private ContentMappings(@Nonnull XmlProxy proxy) {
             stopWords = proxy.stopWords == null ? getDefaultStopWords() : proxy.stopWords;
             mappings = proxy.stopWords == null
    -                ? new ConcurrentHashMap<>()
    -                : proxy.mappings.stream().collect(toConcurrentMap(ContentMapping::getOriginal, Function.identity(), (a, b) -> {throw new IllegalArgumentException();}, ConcurrentSkipListMap::new));
    +                ? new ConcurrentSkipListMap<>(COMPARATOR)
    +                : proxy.mappings.stream().collect(toConcurrentMap(ContentMapping::getOriginal, Function.identity(), (a, b) -> {throw new IllegalArgumentException();}, () -> new ConcurrentSkipListMap(COMPARATOR)));
         }

Reply:
Good catch. I've added a test for this as well by verifying the ordering of mappings created when creating some freestyle projects.